### PR TITLE
set createdFromBytes also when using decodeByteArray with options

### PR DESCRIPTION
**Pull request by edwinvanderham**

### CODE REVIEW

File: shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmapFactory.java [hunk 1]
```diff
@@ -225,6 +225,8 @@ protected static Bitmap decodeByteArray(
       return null;
     }
     Bitmap bitmap = create(desc, opts, image);
+    ShadowBitmap shadowBitmap = Shadow.extract(bitmap);
+    shadowBitmap.createdFromBytes = data;
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmapFactory.java:229]: What about removing `shadowBitmap.createdFromBytes = data` from https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmapFactory.java#L204-L205? Another `decodeByteArray` calls the variant you modified.
And could you help to add related tests for it?

**Review by utzcoz — CHANGES_REQUESTED** [review]

File: robolectric/src/test/java/org/robolectric/shadows/ShadowBitmapFactoryTest.java [hunk 2]
```diff
@@ -142,6 +142,18 @@ public void decodeBytes_shouldSetDescriptionAndCreatedFrom() throws Exception {
     assertEquals(100, bitmap.getHeight());
   }
 
+  @Test
+  public void decodeBytesWithOptions_shouldSetDescriptionAndCreatedFrom() throws Exception {
```
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowBitmapFactoryTest.java:146]: What about changing name to `decodeBytes_shouldSetDescriptionAndCreatedFromWithOptions` and squashing all commits to one?

Comment (edwinvanderham) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowBitmapFactoryTest.java]: @utzcoz I updated the name, I am not sure how to configure this PR to be squashed into a single commit. I normally use gitlab.

**Comment by utzcoz** [comment]
@edwinvanderham could you help to squash all commits to one?

**Comment by edwinvanderham** [comment]
> @edwinvanderham could you help to squash all commits to one?

Could you please explain how I can do this?

**Comment by utzcoz** [comment]
> > @edwinvanderham could you help to squash all commits to one?
> 
> Could you please explain how I can do this?

There is a tutorial to explain how to squash commits: https://levelup.gitconnected.com/how-to-squash-git-commits-9a095c1bc1fc.

**Review by utzcoz — APPROVED** [review]

**Comment by utzcoz** [comment]
Hi @hoisie , could you help to review this PR? Thanks.

**Review by hoisie — APPROVED** [review]

**Comment by utzcoz** [comment]
@edwinvanderham Thanks.

**Comment by edwinvanderham** [comment]
@utzcoz thank you too, this will help me get rid of some unnecessary code in my own custom shadow implementation that we use for testing our software

**Comment by utzcoz** [comment]
> @utzcoz thank you too, this will help me get rid of some unnecessary code in my own custom shadow implementation that we use for testing our software

If you have other improvements for shadow or other parts, we're very welcome to contribute back. :)

**Pull request merged**